iiod-responder.c

```diff
@@ -414,33 +421,47 @@ bool iiod_io_has_response(struct iiod_io *io)
 {
 	uint64_t timeout_us = io->timeout_ms * 1000;
 
-	return io->r_done ||
-		(read_counter_us() - io->w_io.start_time > timeout_us);
+	if (io->r_done)
+		return true;
+
+	if (!io->timeout_ms)
+		return false;
+
+	timeout_us = io->timeout_ms * 1000;
+
+	return read_counter_us() - io->w_io.start_time > timeout_us;
+}
+
+static int iiod_io_cond_wait(struct iiod_io *io)
+{
+	uint64_t diff_ms, timeout_ms = io->timeout_ms;
+
+	if (!timeout_ms)
+		return iio_cond_wait(io->cond, io->lock, 0);
+
+	diff_ms = (read_counter_us() - io->r_io.start_time) / 1000;
+
+	if (diff_ms < timeout_ms)
+		return iio_cond_wait(io->cond, io->lock, timeout_ms - diff_ms);
+
+	return -ETIMEDOUT;
 }
 
 intptr_t iiod_io_wait_for_response(struct iiod_io *io)
 {
 	struct iiod_responder *priv = io->responder;
-	uint64_t diff_ms, timeout_ms = io->timeout_ms;
 	int ret = 0;
 
 	iio_mutex_lock(io->lock);
 
 	while (!io->r_done) {
-		diff_ms = (read_counter_us() - io->r_io.start_time) / 1000;
-
-		if (diff_ms < timeout_ms)
-			ret = iio_cond_wait(io->cond, io->lock, timeout_ms - diff_ms);
-		else
-			ret = -ETIMEDOUT;
-
+		ret = iiod_io_cond_wait(io);
```

Comment:
Hmmm, not a big deal as the change is not that big but maybe we should introduce the `iiod_io_cond_wait()` helper in it's own patch?

Reply:
Ok.